### ISSUE TITLE
FTL DNS Service keeps crashing

### ISSUE THREAD

**Issue by hmlendea** [pi-hole/FTL#662]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate 

**How familiar are you with the codebase?:**

1

---

**[BUG | ISSUE] Expected Behaviour:**

FTL Service starts sucessfully and does its job

**[BUG | ISSUE] Actual Behaviour:**

FTL Service fails to start repeatedly

**[BUG | ISSUE] Steps to reproduce:**

- Start/Restart the DNS via systemd and/or web UI

**Log file output [if available]**

```
[2019-12-12 10:29:16.979 8247] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2019-12-12 10:29:16.979 8247] ---------------------------->  FTL crashed!  <----------------------------
[2019-12-12 10:29:16.979 8247] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2019-12-12 10:29:16.979 8247] Please report a bug at https://github.com/pi-hole/FTL/issues
[2019-12-12 10:29:16.979 8247] and include in your report already the following details:
[2019-12-12 10:29:16.979 8247] FTL has been running for 0 seconds
[2019-12-12 10:29:16.979 8247] FTL branch: master
[2019-12-12 10:29:16.979 8247] FTL version: v4.3.1
[2019-12-12 10:29:16.980 8247] FTL commit: 
[2019-12-12 10:29:16.980 8247] FTL date: 2019-07-08
[2019-12-12 10:29:16.980 8247] FTL user: started as pihole, ended as pihole
[2019-12-12 10:29:16.980 8247] Received signal: Segmentation fault
[2019-12-12 10:29:16.980 8247]      at address: 1681078114
[2019-12-12 10:29:16.980 8247]      with code: SEGV_MAPERR (Address not mapped to object)
[2019-12-12 10:29:16.981 8247] Backtrace:
[2019-12-12 10:29:16.981 8247] B[0000]: /usr/bin/pihole-FTL(+0xf22c) [0x4af22c]
[2019-12-12 10:29:16.981 8247] B[0001]: /usr/lib/libc.so.6(__default_rt_sa_restorer+0) [0xb6d402f0]
[2019-12-12 10:29:16.981 8247] B[0002]: /usr/bin/pihole-FTL(sqlite3_vmprintf+0x24) [0x540f90]
[2019-12-12 10:29:16.981 8247] B[0003]: /usr/bin/pihole-FTL(sqlite3_mprintf+0x4c) [0x54105c]
[2019-12-12 10:29:16.981 8247] B[0004]: /usr/bin/pihole-FTL(sqlite3_overload_function+0x8c) [0x55747c]
[2019-12-12 10:29:16.981 8247] B[0005]: /usr/bin/pihole-FTL(+0xcfe98) [0x56fe98]
[2019-12-12 10:29:16.981 8247] B[0006]: /usr/bin/pihole-FTL(db_init+0x78) [0x4b449c]
[2019-12-12 10:29:16.981 8247] B[0007]: /usr/bin/pihole-FTL(main+0x174) [0x4ad6ac]
[2019-12-12 10:29:16.981 8247] B[0008]: /usr/lib/libc.so.6(__libc_start_main+0x110) [0xb6d2acc0]
[2019-12-12 10:29:16.982 8247] Thank you for helping us to improve our FTL engine!
[2019-12-12 10:29:16.982 8247] FTL terminated!
```

**Device specifics**

Hardware Type: Raspberry Pi 4 with 4GB RAM
OS: Arch Linux ARM

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by DL6ER** [comment]
This is something we have not seen before.

Does this always happen?

What is the output of
```
addr2line -e /usr/bin/pihole-FTL 0xd6ac 0x1449c 0xcfe98 0xb747c 0xa105c 0xa0f90 0xf22c
```
?

**Comment by hmlendea** [comment]
Sorry for the late reply, I was away.

This is the result:
`
??:?
??:?
??:?
??:?
??:?
??:?
??:?
`

**Comment by hmlendea** [comment]
This morning I had the pleasant surprise of ssh-ing into a silky smooth connection (feels good).
So I'm here now to report on my latest findings.

TL;DR: The problem is gone after a reinstall.

The FTL was still down even tho the internet connection was fine, so I concluded it was either a problem with FTL or my system.
I caved and decided to reinstall PiHole completely, even at the risk of loosing the history and configuration (better than having one that doesn't work).
I had a lot of trouble installing FTL again. Apparently, I was missing the some packages in the `base-devel` group (probably automake), and the make command was complaining that it was not inside a git repository (Makefile calls git in the first few lines). After installing everything in base-devel (surely it was overkill but at this point I don't mind that), the make command ran successfully, and the rest of the installation had no issues.

After that, the web interface reported no issue with FTL 

I am sorry tho that we didn't find exactly what happened and caused the issue that I've had.
In light of not having enough evidence, I'm going to assume that something got corrupted on my system.

**Comment by DL6ER** [comment]
We're not really able to give Arch Linux support. Arch Linux support is given by the Arch Linux community themselves as we have no developer running Arch Linux and, hence, are complelty lacking any experience with this OS. Our officially supported distributions are Debian + derivatives (Raspbian, Ubuntu, Linux Mint, ...) and Fedora/CentOS. The Arch Linux community has some ver reliable and responsive people that can usually help out with all issues.

I'm inclined to agree with you that this seems to have been a system error. Especially also because the `addr2line` command came back with some totally useless result which is something we haven't seen before, either. Your assumption that some update might broke something might be true. Even though FTL is (in it's non-patched source code variant) configured to use static libraries wherever possible, it may be that the Arch Linux patches make it use shared libraries (which is not a bad thing, every side has its ups and downs) and then an update *could* have caused an issue.

**Comment by DL6ER** [comment]
I will close this as it seems to be an Arch Linux dependent issue we cannot reproduce any more.
